src/uri/fetchers/docker.cpp

```diff
@@ -268,6 +276,13 @@ static Future<int> download(
   argv.push_back("--connect-timeout");
   argv.push_back("30");
 
+  // to avoid introducing a new option, we simply reuse stall_timeout option
+  // for container image layers we simply multiply this value by a constant factor
+  if (stallTimeout.isSome()) {
+    argv.push_back("--max-time");
+    argv.push_back(std::to_string(5 * static_cast<long>(stallTimeout->secs())));
```

Comment:
5 sec to fetch a docker image, sounds a bit small to me no?
Just DNS might (with retry) reach quickly 5s...

You might maybe use `--connect-timeout 5` and --max-time something_bigger, what do you think?

Reply:
stallTimeout value is 60s by default, to it gives 5minutes per layer (which is higher than the executor_registration_timeout)

Reply:
huh, ok, sorry, make sense, (I did not see stallTimeout value and wrongly interpreted it as 5s)